conn.go

```diff
@@ -90,8 +93,13 @@ func NewConn(addr string, config *Config, delegate ConnDelegate) *Conn {
 	if !config.initialized {
 		panic("Config must be created with NewConfig()")
 	}
+	socketType := "tcp"
```

Comment:
Probably, it's better to organise this part as method `func (*Conn) socketType() string` instead of procedure since it won't be called too frequently, but in general LGTM.

Reply:
Fixed, thank you!